### DISCUSSION TITLE
Setting `ScrollViewer.HorizontalScrollBarVisibility` of a `ListBox` without XAML?

### DISCUSSION THREAD

**Discussion by AzuxirenLeadGuy** [AvaloniaUI/Avalonia#12029] [Q&A] [answered]
Is there any way to  set `ScrollViewer.HorizontalScrollBarVisibility` of a `ListBox` in a C# program, without using XAML?

**Comment by rabbitism** [comment]
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ScrollViewer.cs#L464

Use this method.

**Reply by AzuxirenLeadGuy** [reply]
Hi, thank you for your reply. The method you showed is of ScrollViewer itself. How can I get the Scrollviewer component of the ListBox as I asked in the question?

**Reply by timunie** [reply]
use VisualTree extension methods to get child of type scrollviewer.

**Reply by AzuxirenLeadGuy** [reply]
Thanks for your reply @rabbitism, @timunie . I tried this snippet of code
```csharp

LinkList = new ListBox()
{
     .... // Initialized with items
};
var scroll = Avalonia.VisualTree.VisualExtensions.GetVisualChildren(LinkList).OfType<ScrollViewer>().FirstOrDefault();
if (scroll == null)
{
	System.Console.WriteLine("Invalid scroll!");
}
else
{
        scroll.HorizontalScrollBarVisibility = Avalonia.Controls.Primitives.ScrollBarVisibility.Disabled;
	scroll.VerticalScrollBarVisibility = Avalonia.Controls.Primitives.ScrollBarVisibility.Disabled;
}
```

Unfortunately, the `scroll` returned here is null, since "Invalid scroll!" is printed in the console. What am I doing wrong here? Apologies if I'm being dumb here.


I should probably explain my real issue here. I have a ScrollViewer that contains a ListBox. On building the app, I get two vertical scrolls (one for the ListBox, and one for the outer ScrollViewer); I want only a single scroll, so I am trying to disable the ListBox's scroll.

**Comment by rabbitism** [comment]
If you are using FluentTheme, the inner scrollviewer property is already bound to the attached property of ListBox.
https://github.com/AvaloniaUI/Avalonia/blob/6705d3836634fdb73c92e0e55cdd25e548a024b1/src/Avalonia.Themes.Fluent/Controls/ListBox.xaml#L36

So setting it on ListBox will direct affect the inner scrollviewer.

**Reply by AzuxirenLeadGuy** [reply]
I am trying to access this property from C# code, rather than XAML

**Reply by rabbitism** [reply, marked as answer]
Seems like you don't understand how attached property works. 
ListBox default theme is designed that the inner scrollbar visibility is bound to the attached property of parent ListBox. 
So you set the attached property on ListBox will affect the inner scrollbar visibility.

So you can just use

```
ScrollViewer.SetHorizontalScrollBarVisibility(yourListBox, false);
```

that will change the scrollbar visibility of `yourListBox`'s inner scrollviewer

**Reply by AzuxirenLeadGuy** [reply]
Ah you're right, I didn't understand this till now. I'm a bit new to this, so thank you for being patient with me.
